block-node/publisher/src/main/java/org/hiero/block/node/publisher/BlockStreamProducerSession.java

```diff
@@ -240,7 +249,7 @@ public void onNext(List<BlockItemUnparsed> items) throws RuntimeException {
             // update the live block items received metric
             liveBlockItemsReceived.add(items.size());
             // check items to see if we are entering a new block
-            final boolean newBlock = items.size() == 1 && items.getFirst().hasBlockHeader();
+            final boolean newBlock = items.getFirst().hasBlockHeader();
```

Comment:
We should handle case items is empty

Reply:
Should probably cover the empty case in tests as well. Think it should not ever happen but we need to handle with something better than `NoSuchElementException` from `getFirst()`